### PR TITLE
feat(vue-app): universal fetch

### PR DESCRIPTION
**Pull request by pi0**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
Add universal `fetch` support to Nuxt.js 💥  (Original idea credits: @Atinux)

Users can control over disabling externals using the new `fetch` option. Defaults:
```js
{
  fetch: {
    server: true,
    client: true
  }
}
```

### Polyfill Overhead

Tested on `basic` fixture

#### Bundle Analyzer

Gzip: `78.65 KiB` ~> `79.42 KiB` (`0.77 KiB`)
Normal: `216.73 KiB` ~> `219.17 KiB` (`+2.44 KiB`)

#### `app` Chunk Size

- Server: `31.3 KiB` ~> `31.4 KiB`
- Client: `41.7 KiB` ~> `44.1 KiB`

### Install Size Additions

For `nuxt-start` we have to add `node-fetch` because of externalization.

- `unfetch`: [install size](https://packagephobia.now.sh/result?p=unfetch)
- `node-fetch`: [install size](https://packagephobia.now.sh/result?p=node-fetch)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/config/src/config/_app.js [hunk 1]
```diff
@@ -13,6 +13,11 @@ export default () => ({
     script: []
   },
 
+  fetch: {
+    server: true,
```
Comment (clarkdo) [inline packages/config/src/config/_app.js]: A good way for excluding package from bundles

**Review by galvez — APPROVED** [review]

**Review by atinux — APPROVED** [review]

**Comment by husayt** [comment]
I guess, bye-bye axios

**Comment by pi0** [comment]
@husayt no actually. Axios still rocks but we have even better plans 😊

**Comment by husayt** [comment]
I am intrigued now. At least give a clue, please

**Comment by husayt** [comment]
It is now asking for node-fetch to be installed

> ERROR  Please install missing dependencies:
>npm i node-fetch@^2.3.0

**Comment by pi0** [comment]
@husayt It is in public beta now: https://github.com/nuxt/http :)

**Comment by pi0** [comment]
Regarding the error, if other dependency installs 1.x version of node-fetch Nuxt.js detects it and requires manually installing to the root of the project to prevent inconsistencies.

**Comment by kingda** [comment]
how can i find the example to use it ?

**Comment by pi0** [comment]
@kingda, it is simply fetch() function usable in asyncData and components

https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API

**Pull request merged**